### ISSUE TITLE
Scan parquet with limit by groups

### ISSUE THREAD

**Issue by s-b90** [pola-rs/polars#10291]
### Problem description

For instance, we have a df saved as parquet:
type | version | value
-- | -- | --
str | u16 | u32
a | 1 | 10
a | 1 | 15
b | 3 | 20
a | 1 | 8
a | 2 | 11

When read from this file we may need only first 2 values for each of group (type & version) and in this case we do not need to read whole df in memory.
E.g.
```pl.scan_parquet('path_to_file', limit_by_group=(2, ['type', 'version']))```
type | version | value
-- | -- | --
str | u16 | u32
a | 1 | 10
a | 1 | 15
b | 3 | 20
a | 2 | 11

So basically it should work like ```pl.scan_parquet('path_to_file').groupby(['type', 'version']).head(2)``` just with pushing limitation on scan part.

Also, it may be useful to have a limit_by_group method for dataframe/series.
Maybe related to #10293

**Comment by s-b90** [comment]
#10293

**Comment by orlp** [comment]
As I said on the related issue, this optimization should not be a user-facing option, and should automatically be handled by the query engine. Polars already does quite a few optimizations like this so we'd need to see if it's not already handled and what's needed to make it so.

**Comment by orlp** [comment]
Thinking about it some more, this optimization also doesn't really make much sense in full generality. There is no guarantee that parquet stores your data grouped in the same way as your query asks for. Either way this is not something we wish to explicitly support by user directive, this will be part of further query engine optimization under-the-hood.
